Quadriga/src/main/java/edu/asu/spring/quadriga/rest/ConceptCollectionRestController.java

```diff
@@ -73,425 +75,418 @@
+        } catch (ResourceNotFoundException e) {
+            logger.error("Exception:", e);
+            throw new RestException(404);
+        } catch (ParseErrorException e) {
+
+            logger.error("Exception:", e);
+            throw new RestException(404);
+        } catch (MethodInvocationException e) {
+
+            logger.error("Exception:", e);
+            throw new RestException(403);
+        } catch (QuadrigaStorageException e) {
+            logger.error("Exception:", e);
+            throw new RestException(500);
+        } catch (Exception e) {
+            logger.error("Exception:", e);
+            throw new RestException(403);
+        }
+
+    }
+
+    /**
+     * Rest interface for uploading XML for concept collection http://<<URL>:
+     * <PORT>>/quadriga/rest/syncconcepts/{conceptCollectionID}
+     * hhttp://localhost:8080/quadriga/rest/syncconcepts/
+     * 
+     * @author Lohith Dwaraka
+     * @param request
+     *            {@link HttpServletRequest} object to access request URL param
+     * @param response
+     *            {@link HttpServletResponse} object to access response and
+     *            setting status or/and header details
+     * @param xml
+     *            POST data in terms of XML from the client
+     * @param accept
+     *            Accepts data of type Application/XML
+     * @return Returns of Application/XML to the client. XML contains concept
+     *         which client doesn't have. It helps in syncing the concept at
+     *         client and Quadriga.
+     * @throws QuadrigaException
+     * @throws IOException
+     * @throws SAXException
+     * @throws ParserConfigurationException
+     * @throws JAXBException
+     * @throws QuadrigaAccessException
+     * @throws QuadrigaStorageException
+     * @throws RestException
+     */
+    @RestAccessPolicies({
+            @ElementAccessPolicy(type = CheckedElementType.CONCEPTCOLLECTION_REST, paramIndex = 1, userRole = {
+                    RoleNames.ROLE_CC_COLLABORATOR_ADMIN, RoleNames.ROLE_CC_COLLABORATOR_READ_WRITE }) })
+    @RequestMapping(value = "rest/syncconcepts/{conceptCollectionID}", method = RequestMethod.POST)
+    public ResponseEntity<String> addConceptsToConceptColleciton(
+            @PathVariable("conceptCollectionID") String conceptCollectionId, HttpServletRequest request,
+            HttpServletResponse response, @RequestBody String xml, @RequestHeader("Accept") String accept,
+            Principal principal) throws QuadrigaException, ParserConfigurationException, SAXException, IOException,
+                    JAXBException, QuadrigaAccessException, QuadrigaStorageException, RestException {
+        IUser user = userManager.getUser(principal.getName());
+        if (xml.equals("")) {
+            String errorMsg = restMessage.getErrorMsg("Please provide XML in body of the post request.", request);
+            return new ResponseEntity<String>(errorMsg, HttpStatus.NOT_FOUND);
+        } else {

```

Comment:
loose this else. It's unnecessary. the return statement befreo will make sure, the code after the if will never be reached.